pkg/splunk/enterprise/util.go

```diff
@@ -211,6 +211,85 @@ func GetSmartstoreRemoteVolumeSecrets(volume enterprisev1.VolumeSpec, client spl
 	return accessKey, secretKey, namespaceScopedSecret.ResourceVersion, nil
 }
 
+// ApplyAppListingConfigMap creates the configMap  with two entries:
+// (1) app-list-local.yaml
+// (2) app-list-cluster.yaml
+func ApplyAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject,
+	appConf *enterprisev1.AppFrameworkSpec, appsSrcDeployStatus map[string]enterprisev1.AppSrcDeployInfo) (*corev1.ConfigMap, bool, error) {
+
+	var err error
+	var crKind string
+	var configMapDataChanged bool
+	crKind = cr.GetObjectKind().GroupVersionKind().Kind
+
+	scopedLog := log.WithName("ApplyAppListingConfigMap").WithValues("kind", crKind, "name", cr.GetName(), "namespace", cr.GetNamespace())
+
+	mapAppListing := make(map[string]string)
+
+	// ToDo: sgontla/Jeff: Adapt to any Ansible changes(if required) for differentiating local vs. cluster scope (?)
+	yamlConfHeader := fmt.Sprintf(`splunk:
+  apps_location:`)
+
+	var localAppsConf, clusterAppsConf string
+	localAppsConf = yamlConfHeader
+	clusterAppsConf = yamlConfHeader
+
+	for appSrc, appSrcDeployInfo := range appsSrcDeployStatus {
+		appDeployList := appSrcDeployInfo.AppDeploymentInfoList
+
+		switch scope := getAppSrcScope(appConf, appSrc); scope {
+		case "local":
+			for idx := range appDeployList {
+				if appDeployList[idx].DeployStatus == enterprisev1.DeployStatusPending {
+					localAppsConf = fmt.Sprintf(`%s
+    - "/init-apps/%s"`, localAppsConf, appDeployList[idx].AppName)
```

Comment:
Does this create the correct yml?  Ideally the yml for each scope should look like this:
```
splunk:
  apps_location:
    - "/init-apps/appSrc1/app1.tgz"
    - "/init-apps/appSrc1/app2.tgz"
    - "/init-apps/appSrc2/app3.tgz"
```
Did this format of string work for ansible to install the apps?

Reply:
I believe you, was just making sure :)

Reply:
I have to correct my last statement. My logic assumes we write to "/init-apps/" as a flat directory. addressing the change to append the app source name.